tensorflow/compiler/xla/service/gpu/cudnn_conv_runner.cc

```diff
@@ -223,7 +223,16 @@ Status RunCudnnConvImpl(const CudnnConvParams& params,
   auto output_buf = se::DeviceMemory<OutputType>(params.output_buf);
   AlgorithmConfig algorithm = params.algorithm;
 
-  if (options.algo_override) {
+  // in ROCm mode, the first call to run the convolution needs to trigger the
```

Comment:
I don't quite understand the logic here; but it seems very strange that CUDA-specific file has ROCM-specific workaround?

Reply:
This file is shared across `ROCm` and `Cuda`. I can open a new PR after this one with renaming changes. 

Regarding the workaround, it has a long history here. If I remember it right, the check for whether algorithm config is empty is [here](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/stream_executor/rocm/rocm_dnn.cc#L2623).